Auth/EnvironmentSelect.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { Modal, RadioButton } from 'react-native-paper';
+import { APP_ENVIRONMENTS, useAppEnvironment } from '../AppEnvironment';
+import RoundedButton from '../Components/RoundedButton';
+
+type EnvironmentSelectProps = {
+  visible: boolean;
+  onDismiss: () => void;
+};
+
+export default function EnvironmentSelect({
+  visible,
+  onDismiss,
+}: EnvironmentSelectProps) {
+  
+  const {environment, setEnvironment} = useAppEnvironment();
+  const [selectedEnv, setSelectedEnv] = useState<string>(environment.name.toLowerCase());
+
+    return (
+      <Modal visible={visible} onDismiss={onDismiss} contentContainerStyle={styles.sheet}>
+          <Text style={styles.title}>Change Server</Text>
+
+          <RadioButton.Group onValueChange={setSelectedEnv} value={selectedEnv}>
```

Comment:
If possible, I would suggest making this a free-text input, so we can more easily point it to e.g. local test servers for development.

Reply:
Yes, that's the end goal for fetching the client ID at runtime, so you can point the app to any server and it just works